src/components/data-pane/data-selector.tsx

```diff
@@ -188,9 +188,10 @@ const DataSelectorRenderer = CSSModules(DataSelectorBase, styles);
 
 export const DataSelector = connect(
   (state: State) => {
+    const presentUndoableState = state.undoable.present;
     return {
-      data: state.present.dataset,
-      name: state.present.dataset.name,
+      data: presentUndoableState.dataset,
```

Comment:
We should use a selector for dataset for consistency and for reduce the amount of updates needed if we reshape the state in the future.